### ISSUE TITLE
Using script with Automate?

### ISSUE THREAD

**Issue by jackorama** [hsw107/Photoshop-Export-Layers-to-Files-Fast#90]
I'm afraid this is a really stupid question, but...

Can your script be used in tandem with Photoshop's Automate? I put the script in /Applications/../Presets/Scripts/, and it now appears in the Scripts submenu. 

I include it in an Automate action successfully. But when I run the action, it stops on the main window (for some reason it's not recording the fact that I clicked on the "Run" button in the action). It's stopping [here](http://bit.ly/29lUylq).

I would love to be able to set it up to run this Export script across several PSDs at once, without me being around. Your script is amazing and so much faster than Layers To Files...

Any help would be hugely appreciated.

**Comment by JosephMerc** [comment]
I know this is an old post, but I was wondering if there was any traction on this. This is exactly what I need to finish the the full automation of a process of mine.

**Comment by hsw107** [comment]
@JosephMerc Unfortunately there's no "team" working on this. It's simply open source and as people have time and interest to pick up certain features, they happen.
Personally I'd love to do this but right now it sounds like a lot of work and I don't have time to get into it at the moment and adobe documentation on scripting can be rather hard to parse.

**Comment by JosephMerc** [comment]
@hsw107 Thanks for the quick reply! 

That makes definite sense. I'll do what I can to figure out how to move forward, but if anyone here has any recommendations on how to make this work, I'd love to hear them. I'm new to Javascript, but I'm exploring triggering some key presses that cycle over to run and trigger the dialogue once it's popped up.

**Comment by JosephMerc** [comment]
Hey! For anyone who is looking to automate this, I've found a workaround using another program. If you set up a persistent script in Autohotkey (or similar program), you can have a script that identifies when the dialogue box pops up, and trigger an action that tabs through the box and runs this script. 

I was thrilled to have found this! I can share more info if anyone is interested.

**Comment by daniellewooding** [comment]
> Hey! For anyone who is looking to automate this, I've found a workaround using another program. If you set up a persistent script in Autohotkey (or similar program), you can have a script that identifies when the dialogue box pops up, and trigger an action that tabs through the box and runs this script.
> 
> I was thrilled to have found this! I can share more info if anyone is interested.

@JosephMerc I know it's been a while, but if you still have the AutoHotkey solution for this I would be very interested.

@hsw107 Was there ever a solution built-in for this? Similar to @JosephMerc I'm really finding a lot of benefit to your script, but I'm attempting to use it in a series of actions that will save and export hundreds of combinations of edited layers and the only thing preventing me from setting it to run overnight to set-it-and-forget-it is that for each export I need to press the run button, and the confirmation button when it ends.

**Comment by JosephMerc** [comment]
Oh man, it's been a while since I've been on this thread; I want to send a big thank you to @hsw107 for this script. It's literally been a life changer, and I've learned enough javascript to start elaborating on things like how PNGs get are named when they're exported, and things like that. It's been a wonderful trip, so thank you for the ewonderful script!

@daniellewooding , I do have the script kicking around still! You'll need to install Autohotkey, which should be a quick and easy install, once you find it online. Once you do, you'll create a text file in Notepad, and pop the code below in, and rename the ".txt" extension to ".ahk", which should change the icon to something like a large "H", if Autohotkey was installed properly (Or download the txt file I've attached, and rewrite the extension as ".ahk")

[AutoPNG.txt](https://github.com/hsw107/Photoshop-Export-Layers-to-Files-Fast/files/7017770/AutoPNG.txt)

:

`#Persistent

SetTitleMatchMode 2
SetTimer, RunPNG, 2000

Return

RunPNG:

IfWinExist, Export Layers To Files
{
  Send {rshift Down}{TAB 4}{rshift Up}
  Send {space}
  return
}
return
`

What the AHK script does is essentially identifies when the Export diaologue box of hsw107's script is present on screen, and (simply enough) just tabs through the buttons to "Run", and then simulates pressing the space bar to run the script. 

It has saved me a lot of tedious work when it comes to batch processing PSDs in photoshop. 

Some things to note; 

1. It's best not to touch your computer while the script is running; since this simulates keystrokes, if you start changing the focus on your screen, the script will start incessantly tabbing through whatever other window you've selected and act a nuisance. Ideally, you set up your Batch, and you let it run until it's done. 
2. Before you activate the AHK script, manually set up the settings for and export your first image with hsw107's script; after you've manually hit "Run" and are exporting, that's when you activate the AHK script and it'll take care of all the other images that your batch brings up to process. If you have the AHK script running right from the get go, your image will start exporting before you even have a chance to change the export destination. Furthermore, the AHK script (currently) isn't smart enough to know if you've changed the focus on the dialogue box; it only knows how to Tab through  to the Run button from the default focus the window has when it first appears; so not great if you're changing export settings on the first image of a new batch, but works perfectly for every image after that. 
3. Get familiar with how to end or pause an AHK script, if you're not already; you'll want to close the script between separate batches, and you'll want to be able to quickly turn the script off if it starts tabbing through things it shouldn't. 

Feel free to try it out and let me know if you run into trouble; I'll do what I can to help out!

**Comment by JosephMerc** [comment]
@daniellewooding That at least covers getting the script to run without manually hitting the Run button. 
However, I don't recall the confirmation at the end you mention needing to push; it's possible I found a way to nix that ages back. I'd have to double check, and try exporting with the original script again.

**Comment by hsw107** [comment]
Hi there! With version v2.4.0 I added a "Run Silently" checkbox which removes the final confirmation message. Hope this helps!
